pkg/noderesourcetopology/filter.go

```diff
@@ -0,0 +1,263 @@
+			// memory will not be provided in CRD
+			// 2. set numa node as possible node if resource is cpu and it's not guaranteed QoS, since cpu will flow
+			// 3. set numa node as possible node if zero quantity for non existing resource was requested (TODO check topology manaager behaviour)
+			// 4. otherwise check amount of resources
+			if resource == v1.ResourceMemory ||
+				strings.HasPrefix(string(resource), string(v1.ResourceHugePagesPrefix)) ||
+				resource == v1.ResourceCPU && qos != v1.PodQOSGuaranteed ||
+				quantity.Cmp(zeroQuantity) == 0 ||
+				numaQuantity.Cmp(quantity) >= 0 {
+				resourceBitmask.Add(numaNode.NUMAID)
+			}
+		}
+		bitmask.And(resourceBitmask)
+	}
+}
+
+func SingleNUMAPodLevelHandler(pod *v1.Pod, zones topologyv1alpha1.ZoneList) *framework.Status {
+	klog.V(5).Infof("Pod Level Resource handler")
+	containers := []v1.Container{}
+	containers = append(pod.Spec.InitContainers, pod.Spec.Containers...)
+
+	resources := make(v1.ResourceList)
+
+	for _, container := range containers {
+		for resource, quantity := range container.Resources.Requests {
+			if quan, ok := resources[resource]; ok {
+				quantity.Add(quan)
+			}
+			resources[resource] = quantity
+		}
+	}
+
+	nodes := createNUMANodeList(zones)
+	bitmask := bm.NewEmptyBitMask()
+	bitmask.Fill()
+	checkResourcesForNUMANodes(bitmask, nodes, resources, v1qos.GetPodQOS(pod))
+
+	if bitmask.IsEmpty() {
+		// definitly we can't align container, so we can't align a pod
+		return framework.NewStatus(framework.Unschedulable, fmt.Sprintf("Cannot align pod: %s", pod.Name))
+	}
+	return nil
+
+}
+
+func createNUMANodeList(zones topologyv1alpha1.ZoneList) NUMANodeList {
+	nodes := make(NUMANodeList, 0)
+	for _, zone := range zones {
+		if zone.Type == "Node" {
+			var numaID int
+			fmt.Sscanf(zone.Name, "node-%d", &numaID)
+			resources := extractResources(zone)
+			nodes = append(nodes, NUMANode{NUMAID: numaID, Resources: resources})
+		}
+	}
+	return nodes
+}
+
+func getNodeTopology(nodes []*topologyv1alpha1.NodeResourceTopology, nodeName string) *topologyv1alpha1.NodeResourceTopology {
+	for _, node := range nodes {
+		if node.Name == nodeName {
+			return node
+		}
+	}
+	return nil
+}
+
+// Filter Now only single-numa-node supported
+func (tm *TopologyMatch) Filter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
+	if nodeInfo.Node() == nil {
+		return framework.NewStatus(framework.Error, fmt.Sprintf("Node is nil %s", nodeInfo.Node().Name))
+	}
+	if v1qos.GetPodQOS(pod) == v1.PodQOSBestEffort {
+		return nil
+	}
+
+	nodeName := nodeInfo.Node().Name
+	nodeTopology, err := tm.lister.NodeResourceTopologies(tm.namespace).Get(nodeName)
+
+	if err != nil {
+		klog.Errorf("Cannot get NodeTopologies from cache: %v", err)
+		return nil
+	}
+
+	klog.V(5).Infof("nodeTopology: %v", nodeTopology)
+	topologyPolicies := getTopologyPolicies(nodeTopology, nodeName)
+	for _, policyName := range topologyPolicies {
```

Comment:
I think you can direclty use `nodeTopology.TopologyPolicies`? (and cast to the custome string type when searching)

```suggestion
	for _, policyName := range nodeTopology.TopologyPolicies {
```